editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -147,42 +156,48 @@ protected void generateTabContents(List<Parameter> props, final Composite page)
-			txtFieldCreator = createTextFieldWithNoArgMethodBrowse(p, page);
+		} else if (propName.equals(CamelBean.PROP_INIT_METHOD) || propName.equals(CamelBean.PROP_DESTROY_METHOD)) {
 			validator = new PropertyMethodValidator(modelMap, project);
-		} else if (propName.equals(CamelBean.PROP_FACTORY_METHOD)
-				|| propName.equals(CamelBean.PROP_FACTORY_BEAN)) {
-			txtFieldCreator = createTextFieldWithMethodBrowse(p, page);
+		} else if (propName.equals(CamelBean.PROP_FACTORY_METHOD) || propName.equals(CamelBean.PROP_FACTORY_BEAN)) {
 			validator = new PropertyMethodValidator(modelMap, project);
 		} else if (propName.equals(CamelBean.PROP_ID)) {
-			txtFieldCreator = createTextField(p, page);
 			validator = new NewBeanIdPropertyValidator(p, selectedEP);
+		}
+		if (validator == null && p.getRequired() != null && "true".contentEquals(p.getRequired())) { //$NON-NLS-1$
+			validator = new PropertyRequiredValidator(p);
+		}
+		return validator;
+	}
+	
+	private void createPropertyFieldEditor(final Composite page, Parameter p) {
+		IValidator validator = getValidatorForField(p);
+		AbstractParameterPropertyUICreator fieldCreator = null;
+		String propName = p.getName();
+		if (propName.equals(CamelBean.PROP_CLASS)) {
+			fieldCreator = createTextFieldWithClassBrowseAndNew(p, page);
+		} else if (propName.equals(CamelBean.PROP_INIT_METHOD) || propName.equals(CamelBean.PROP_DESTROY_METHOD)) {
+			fieldCreator = createTextFieldWithNoArgMethodBrowse(p, page);
+		} else if (propName.equals(CamelBean.PROP_FACTORY_METHOD) || propName.equals(CamelBean.PROP_FACTORY_BEAN)) {
+			fieldCreator = createTextFieldWithMethodBrowse(p, page);
+		} else if (propName.equals(CamelBean.PROP_ID)) {
+			fieldCreator = createTextField(p, page);
+		} else if (propName.equals(CamelBean.PROP_SCOPE)) {
+			fieldCreator = createScopeCombo(p, page);
 		} else if (CamelComponentUtils.isTextProperty(p) || CamelComponentUtils.isCharProperty(p)) {
-			txtFieldCreator = createTextField(p, page);
-			if (p.getRequired() != null && "true".contentEquals(p.getRequired())) { //$NON-NLS-1$
-				validator = new PropertyRequiredValidator(p);
-			}
-		} else if (CamelComponentUtils.isUnsupportedProperty(p)) {
-			// handle unsupported props
+			fieldCreator = createTextField(p, page);
+		} else if (CamelComponentUtils.isUnsupportedProperty(p)) { // handle unsupported props
 			new UnsupportedParameterPropertyUICreatorForAdvanced(dbc, modelMap, eip, selectedEP, p, page,
 					getWidgetFactory()).create();
 		}
-
-		if (txtFieldCreator != null) {
-			ISWTObservableValue uiObservable = handleObservable(txtFieldCreator, p);
-			if (uiObservable != null) {
-				bindField(validator, uiObservable, p);
-			}
+		if (fieldCreator != null && fieldCreator instanceof AbstractTextFieldParameterPropertyUICreator) {
+			ISWTObservableValue uiObservable = handleObservable((AbstractTextFieldParameterPropertyUICreator) fieldCreator, p);
+			if (uiObservable != null) bindField(validator, uiObservable, p);
```

Comment:
personnally i prefer when it is written on 2 different lines with {} (and i think that Sonar will report it as a minor code smell)
Can we switch back to previous way of writing it?

if you think that we should write it on a single line, might worth discussing it in a specific thread and -when it will be available - to configure in Sonar to ignore this rule